Troubleshooting.md

```diff
@@ -0,0 +1,54 @@
+## Troubleshooting Python 3.12 Virtual Environment Installation and Activation
+
+When installing and managing virtual environments in Ubuntu (and other Linux operating systems), it is important to remember that the OS may require a specific version of python to perform operating system functions. For detection-rules we use Python version 3.12 which has a number of differences between older versions of Python, most notably no longer including `distutils` which can cause some unusual error messages. 
+
+This section of the guide offers some solutions to some common problems that can occur if one inadvertently overwrites the system's Python 3 installation with Python version 3.12. If you are uncertain as to which specific issue you are having, a more generic way to fix the issue is to start with the first issue and work your way through the rest until you no longer have errors. 
```

Comment:
```suggestion
This section of the guide offers some solutions to some common problems that can occur if one inadvertently overwrites the system's Python 3 installation with Python version 3.12. 
```

Reply:
Fine with removing this, but as a note. The typical scenario is that there will be an error when running `make` which will appear potentially unrelated/incomprehensible. More often than not, you will need to perform all of these steps in order without particular error messages directing you that this is what needs to be done.